app/src/main/java/com/example/mytestproject/ui/searchCity/SearchCityViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.example.mytestproject.ui.searchCity
+
+import android.content.Context
+import android.util.Log
+import android.widget.SearchView
+import androidx.lifecycle.ViewModel
+import com.example.data.utils.getCityModelsListFromJson
+import com.example.domain.models.CityModel
+import io.reactivex.Observable
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.Disposable
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import java.util.concurrent.TimeUnit
+
+class SearchCityViewModel : ViewModel() {
+
+    private var disposable: Disposable? = null
+
+    private fun fromView(searchView: SearchView): Observable<String> {
+
+        val subject: PublishSubject<String> = PublishSubject.create()
```

Comment:
Let's not use subjects if we don't need them. You can jut create a Flowable using Flowable.create.
Like here: https://adrianhall.github.io/android/2018/06/20/implementing-search-on-type-in-android-with-rxjava/

Reply:
Fixed